src/processinfo.h

```diff
@@ -216,10 +216,10 @@ class ProcessInfo
     uint64_t _vvarEnd;
     uint64_t _endOfStack;
 
-    uint64_t _clock_gettime_offset;
-    uint64_t _getcpu_offset;
-    uint64_t _gettimeofday_offset;
-    uint64_t _time_offset;
+    ptrdiff_t _clock_gettime_offset;
```

Comment:
Why do we need a signed data type (`ptrdiff_t`) here?

Reply:
CMIIWW,

The function calculate this is `dmtcp_dlsym_lib_fnc_offset`, which could be negative offset I think.

I found this from compile warning, it was also used to compare with a `ptrdiff_t` type variable.

Reply:
The function's offset from the base in the text section would be positive only. You can visualize it like below:
.text base addr|------------| 0x100000
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
function1_addr|------------| 0x100050
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
function2_addr|------------| 0x100080

However, what you are saying is correct that in both places the data type should be consistent. DMTCP's master branch fixes this compiler warning here: https://github.com/dmtcp/dmtcp/commit/c3a6b100315051713f28fa716c53abc16d13fb1f
Following this commit, we'll be aligned to upstream DMTCP.

Reply:
OK, then I will just cherry-pick that one, thanks!